test/infamy/tap.py

```diff
@@ -95,7 +99,10 @@ def succeed(self):
     def skip(self):
         raise TestSkip()
 
-    def fail(self):
+    def fail(self, message=None):
+        if message:
+            self.out.write(f"# {message}\n")
+            self.out.flush()
```

Comment:
Are there any cases where we could not use `assert`s instead? I.e., was the original sin me introducing `fail()` in the first place?

Reply:
Well, even though assert probably is more pythonic, I still think success/fail constitute a clear API.  I'd vote to keep as-is for now, with this addition.